gluten-core/src/main/scala/io/glutenproject/execution/GenerateExecTransformer.scala

```diff
@@ -152,19 +154,65 @@ case class GenerateExecTransformer(
       operatorId: Long,
       inputAttributes: Seq[Attribute],
       input: RelNode,
-      generator: ExpressionNode,
+      generatorNode: ExpressionNode,
       childOutput: JList[ExpressionNode],
       validation: Boolean): RelNode = {
-    if (!validation) {
-      RelBuilder.makeGenerateRel(input, generator, childOutput, context, operatorId)
+    val generateRel = if (!validation) {
+      RelBuilder.makeGenerateRel(input, generatorNode, childOutput, context, operatorId)
     } else {
       // Use a extension node to send the input types through Substrait plan for validation.
       val inputTypeNodeList =
         inputAttributes.map(attr => ConverterUtils.getTypeNode(attr.dataType, attr.nullable)).asJava
       val extensionNode = ExtensionBuilder.makeAdvancedExtension(
         BackendsApiManager.getTransformerApiInstance.packPBMessage(
           TypeBuilder.makeStruct(false, inputTypeNodeList).toProtobuf))
-      RelBuilder.makeGenerateRel(input, generator, childOutput, extensionNode, context, operatorId)
+      RelBuilder.makeGenerateRel(
+        input,
+        generatorNode,
+        childOutput,
+        extensionNode,
+        context,
+        operatorId)
+    }
+    applyPostProjectOnGenerator(generateRel, context, operatorId, childOutput, validation)
+  }
+
+  private def applyPostProjectOnGenerator(
```

Comment:
Could you please add one comment on when this should be called, any other cases than `inline`

Reply:
There are 3 types of CollectionGenerator in spark: `Explode`, `PosExplode` and `Inline`. Looks like only `inline` needs the post projection according Velox `UnnestNode` implementation. Besides, it would be better to implement this function in backends API. I will try `PosExplode` next and refine this part of code in another PR.